static/js/admin/src/AddMenu.js

```diff
@@ -10,51 +10,51 @@ define('AddMenu', [ 'jquery', 'jquery-ui', 'underscore', 'CreateContainerForm',
 	AddMenu.prototype.getMenuItems = function() {
 		var items = {};
 
-		if ($.inArray('addRemoveContents', this.container.permissions) != -1) {
+		if ($.inArray('ingest', this.container.permissions) != -1) {
 			var self = this;
 
 			items["addContainer"] = {
 				name : "Add Container",
 				visible: function(key, opt){
-					return self.isAggregate(self);
+					return self.allowIngestInto(self);
 				}
 			};
 			items["ingestPackage"] = {
 				name : "Add Ingest Package",
 				visible: function(key, opt){
-					return self.isAggregate(self);
+					return self.allowIngestInto(self);
 				}
 			};
 			items["ingestSource"] = {
 				name : "Add from File Server",
 				visible: function(key, opt){
-					return self.isAggregate(self);
+					return self.allowIngestInto(self);
 				}
 			};
 			items["addWork"] = {name : "Add Work",
 				visible: function(key, opt){
-					return self.isAggregate(self);
+					return self.allowIngestInto(self);
 				}
 			};
 			items["addFile"] = {name : "Add File",
 				visible: function(key, opt){
-					if (self.container.type == "Aggregate") {
+					if (self.container.type == "Work") {
 						return true;
 					}
 					return false;
 				}
 			};
 
 		}
-		if ($.inArray('editDescription', this.container.permissions) != -1) {
+		if ($.inArray('bulkUpdateDescription', this.container.permissions) != -1) {
 			items["importMetadata"] = {name : "Import MODS"};
 		}
 		
 		return items;
 	};
 
-	AddMenu.prototype.isAggregate = function(self) {
-		if (self.container.type == "Aggregate") {
+	AddMenu.prototype.allowIngestInto = function(self) {
+		if (self.container.type == "Work" || self.container.type == "File") {
```

Comment:
does this mean it's disallowed to ingest into a work, as well as into a file?

Reply:
Right, this method will block the ingest options from appearing for works and files. Works should only offer "add file" and files can't add any subfiles.

Reply:
ok, thanks for explaining!